nixos/modules/tasks/filesystems/zfs.nix

```
@@ -127,7 +127,8 @@ let
         "systemd-ask-password-console.service"
       ];
       requiredBy = getPoolMounts prefix pool ++ [ "zfs-import.target" ];
-      before = getPoolMounts prefix pool ++ [ "zfs-import.target" ];
+      before = getPoolMounts prefix pool ++ [ "shutdown.target" "zfs-import.target" ];
```

I think this might need parens.
```suggestion
      before = (getPoolMounts prefix pool) ++ [ "shutdown.target" "zfs-import.target" ];
```

---

Nah. It didn't before and it doesn't now. Function application binds stronger than any other syntax.